src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -55,6 +55,15 @@ private async Task ApplyStartContextIfPresent()
 
                 bool success = _instanceManager.StartAssignment(assignmentContext);
                 _logger.LogInformation($"StartAssignment invoked (Success={success})");
+
+                if (assignmentContext.MSIContext != null)
+                {
+                    await _instanceManager.SpecializeMSISidecar(assignmentContext);
+                }
+                else
+                {
+                    _logger.LogInformation("Skipping re-specialization of MSI sidecar.");
```

Comment:
As in the original PR, can't we just leave any required logging in SpecializeMSISidecar?

Reply:
Yes, adding an additional log here lets us query by Source and possibly use it in a detector.
This is an exception scenario and not expected to be hit often.

Reply:
Then I'd say move the null check into SpecializeMSISidecar to centralize it. "Skipping specialization of MSI sidecar - MSI info is empty" or something like that